ArduPlane/commands_logic.cpp

```diff
@@ -613,6 +613,31 @@ bool Plane::verify_takeoff()
     }
 }
 
+/*
+  When enabled, if the previous waypoint is VTOL Takeoff set the flag to do the full rate climbout.
+  This will give a climbing transition and reach the cruise waypoint immediately as opposed to
+  calculating the glide slope based on proportion to the waypoint.
+ */
+#if HAL_QUADPLANE_ENABLED
+void Plane::maybe_do_full_rate_climbout()
```

Comment:
This would probably be neater as `bool Plane::should_do_full_rate_climbout() const` as the early-return when you don't want to do it is trivial.  The caller can `quadplane.set_do_full_rate_climbout(should_do_full_rate_climbout));

Should save a few bytes too.